hi @lukew-cogapp, thanks for the fix! LGTM.

> FAILED tests/polars/test_polars_check.py::test_polars_lazy_failure_cases_nested_column[Array[int, 2]-column1-dtype1-1] - [XPASS(strict)] Custom check signature incompatible with Narwhals backend
FAILED tests/polars/test_polars_check.py::test_polars_lazy_failure_cases_nested_column[List[List[int]]-column4-dtype4-1] - [XPASS(strict)] Custom check signature incompatible with Narwhals backend

Any idea why these are failing? seems like they should be skipped

--- a/tests/polars/test_polars_check.py
+++ b/tests/polars/test_polars_check.py
@@ -320,7 +320,7 @@ def test_polars_dataframe_check_n_failure_cases(lf):
             "Array[int, 2]",
             pl.Series([[1, 2], [3, 4]], dtype=pl.Array(pl.Int64, 2)),
             pl.Array(pl.Int64, 2),
-            "1",
+            "[1,2]",
         ),
         (
             "Struct",
@@ -338,7 +338,7 @@ def test_polars_dataframe_check_n_failure_cases(lf):
             "List[List[int]]",
             pl.Series([[[1, 2]], [[3, 4]]]),
             pl.List(pl.List(pl.Int64)),
-            "1",
+            "[[1,2]]",
         ),
         (
             "Struct[List]",

